gtsfm/ui/gtsfm_process.py

```diff
@@ -0,0 +1,68 @@
+"""
+Defines GTSFMProcess, which holds a UiMetadata object for displaying this process in the process graph.
+
+Author: Kevin Fu
+"""
+
+# When creating a new class, follow this template:
+
+# 1: Import GTSFMProcess, UiMetadata
+# ----------------------------------
+# from gtsfm.ui.gtsfm_process import GTSFMProcess, UiMetadata
+
+# 2: Subclass GTSFMProcess (this can replace ABCMeta, since it inherits from ABCMeta)
+# ------------------------
+# class ClassName(GTSFMProcess):
+
+# 3: Implement get_ui_metadata()
+# ------------------------------
+#    def get_ui_metadata() -> UiMetadata:
+#        """Returns data needed to display this process in the process graph."""
+#
+#        return UiMetadata(
+#                   "Display Name"
+#                   "Parent Plate Name"
+#                   ("Input Product 1", "Input Product 2")
+#                   ("Output Product 1", "Output Product 2")
+#               )
+
+from gtsfm.ui.registry import AbstractableRegistryHolder
+
+import abc
+from dataclasses import dataclass
+
+from typing import Tuple
+
+
+@dataclass(frozen=True, order=True)
+class UiMetadata:
+    """
+    Holds all info needed to display a GTSFMProcess in the process graph (see ProcessGraphGenerator).
```

Comment:
nit: 1-sentence line should go in same line as the triple quote (see style guide)